### PR TITLE
ci: verify changelog entries in CI

### PR DESCRIPTION
**Pull request by Totktonada**
It should make development a bit more friendly: CI will assist with keeping release notes buildable.

The `gen-release-notes` script is written with care to error messages and if it raises an error it should be quite easy to find a cause and fix it. It does not check Markdown syntax and spelling, though: just ability to build the document from files.

The most often mistake is around section header format: `## feature/<...>` or `## bugfix/<...>`.

Just a reminder: if you want to see how your changelog entry will look in a future release notes, just run `./tools/gen-release-notes`. Don't hesitate to the changelog after a push to master: it is information for our users and there is nothing bad in making it better.

### CODE REVIEW

**Comment by Totktonada** [comment]
Should be renewed if PR #6702 will go first.

**Comment by Totktonada** [comment]
Updated after PR #6702 landing, rebased on top of fresh master.

**Review by igormunkin — APPROVED** [review]
@Totktonada, thanks for the patchset! LGTM with a tiny nit for the third patch. Furthermore, I have two general notes to this PR:
* Is it possible to check whether changelog entry is added for the particular series? I don't want to block this changeset with this request now, but it will make maintainer's life easier.
* I'm glad you've mentioned the fact GitHub Actions rules are becoming hell unsupportable from patch to patch. I hope they either add the ability to include some parts of workflow rules the similar way e.g. gitconfig does, or just fix rerun of the particular pipeline in the matrix sooner or later. Otherwise, we need to design our own tooling for pipelines templates.

File: .github/workflows/lint.yml
Comment (Totktonada) [reply .github/workflows/lint.yml:79]: I think that just script name is descriptive enough here.

**Comment by Totktonada** [comment]
> Is it possible to check whether changelog entry is added for the particular series? I don't want to block this changeset with this request now, but it will make maintainer's life easier.

There is no good way to automatically determine, whether a commit need an entry in release notes or not. OTOH, commits that closing issues likely should be reflected somehow. I propose to add a separate check into `lint.yml`.

**Review by locker — APPROVED** [review]

**Pull request merged**
